spec/create_spec.rb

```diff
@@ -124,4 +124,39 @@
 
   end
 
+  context 'validation for AWS credentials' do
+
+    context 'IAM profile is not used' do
+      before do
+        config[:use_iam_profile] = false
+      end
+
+      it 'throws an error when aws key is missing' do
+        config[:aws_access_key_id] = ''
+        expect { driver.send(:aws_credentials) }.to raise_error(Kitchen::UserError)
+      end
+
+      it 'throws an error when aws secret is missing' do
+        config[:aws_secret_access_key] = ''
+        expect { driver.send(:aws_credentials) }.to raise_error(Kitchen::UserError)
+      end
+    end
+
+    context 'IAM profile is used' do
+      before do
+        config[:use_iam_profile] = true
+      end
+
+      it 'throws an error when aws key is missing' do
```

Comment:
description is wrong here, should be "succeeds when using iam profile and aws key is missing" or something